plugins/inputs/beat/beat.go

```diff
@@ -0,0 +1,278 @@
+package beat
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	jsonparser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+const sampleConfig = `
+  ## An URL from which to read Beat-formatted JSON
+  ## Default is "http://127.0.0.1:5066".
+  url = "http://127.0.0.1:5066"
+
+  ## Enable collection of the Beat stats
+  collect_beat_stats = true
+
+  ## Enable the collection if Libbeat stats
+  collect_libbeat_stats = true
+
+  ## Enable the collection of OS level stats
+  collect_system_stats = false
+
+  ## Enable the collection of Filebeat stats
+  collect_filebeat_stats = true
+
+  ## HTTP method
+  # method = "GET"
+
+  ## Optional HTTP headers
+  # headers = {"X-Special-Header" = "Special-Value"}
+
+  ## Override HTTP "Host" header
+  # host_header = "logstash.example.com"
+
+  ## Timeout for HTTP requests
+  timeout = "5s"
+
+  ## Optional HTTP Basic Auth credentials
+  # username = "username"
+  # password = "pa$$word"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+const description = "Read metrics exposed by Beat"
+
+const suffixInfo = "/"
+const suffixStats = "/stats"
+
+type BeatInfo struct {
+	Beat     string `json:"beat"`
+	Hostname string `json:"hostname"`
+	Name     string `json:"name"`
+	UUID     string `json:"uuid"`
+	Version  string `json:"version"`
+}
+
+type BeatStats struct {
+	Beat     map[string]interface{} `json:"beat"`
+	FileBeat interface{}            `json:"filebeat"`
+	Libbeat  interface{}            `json:"libbeat"`
+	System   interface{}            `json:"system"`
+}
+
+type Beat struct {
+	URL string `toml:"url"`
+
+	CollectBeatStats     bool `toml:"collect_beat_stats"`
+	CollectLibbeatStats  bool `toml:"collect_libbeat_stats"`
+	CollectSystemStats   bool `toml:"collect_system_stats"`
+	CollectFilebeatStats bool `toml:"collect_filebeat_stats"`
+
+	Username   string            `toml:"username"`
+	Password   string            `toml:"password"`
+	Method     string            `toml:"method"`
+	Headers    map[string]string `toml:"headers"`
+	HostHeader string            `toml:"host_header"`
+	Timeout    internal.Duration `toml:"timeout"`
+
+	tls.ClientConfig
+	client *http.Client
+}
+
+func NewBeat() *Beat {
+	return &Beat{
+		URL:                  "http://127.0.0.1:5066",
+		CollectBeatStats:     true,
+		CollectLibbeatStats:  true,
+		CollectSystemStats:   true,
+		CollectFilebeatStats: true,
+		Method:               "GET",
+		Headers:              make(map[string]string),
+		HostHeader:           "",
+		Timeout:              internal.Duration{Duration: time.Second * 5},
+	}
+}
```

Comment:
Make this an `Init()` function and later (in `init()`) pass an empty config.

Reply:
Will fix